zipkin/src/main/java/com/linecorp/armeria/client/tracing/HttpTracingClient.java

```diff
@@ -43,38 +45,63 @@
-        return delegate -> new HttpTracingClient(delegate, brave);
+    public static Function<Client<HttpRequest, HttpResponse>, HttpTracingClient> newDecorator(Tracing tracing) {
+        return delegate -> new HttpTracingClient(delegate, tracing);
     }
 
-    HttpTracingClient(Client<HttpRequest, HttpResponse> delegate, Brave brave) {
-        super(delegate, brave);
+    private final Tracer tracer;
+    private final TraceContext.Injector<HttpHeaders> injector;
+
+    /**
+     * Creates a new instance.
+     */
+    protected HttpTracingClient(Client<HttpRequest, HttpResponse> delegate, Tracing tracing) {
+        super(delegate);
+        this.tracer = tracing.tracer();
+        injector = tracing.propagationFactory().create(AsciiStringKeyFactory.INSTANCE)
+                          .injector(HttpHeaders::set);
     }
 
     @Override
-    protected void putTraceData(ClientRequestContext ctx, HttpRequest req, @Nullable SpanId spanId) {
-        final HttpHeaders headers;
-        if (ctx.hasAttr(ClientRequestContext.HTTP_HEADERS)) {
-            headers = ctx.attr(ClientRequestContext.HTTP_HEADERS).get();
-        } else {
-            headers = new DefaultHttpHeaders(true);
-            ctx.attr(ClientRequestContext.HTTP_HEADERS).set(headers);
+    public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Exception {
+        Span span = tracer.nextSpan();
+
+        final String method = ctx.method().name();
+
+        injector.inject(span.context(), req.headers());
+        span.kind(Kind.CLIENT)
+            .name(method)
+            .remoteEndpoint(Endpoint.builder()
+                                    .serviceName(ctx.endpoint().authority())
+                                    .build())
+            .start();
+
+        SpanContextUtil.setupContext(ctx, span, tracer);
+
+        ctx.log().addListener(log -> finishSpan(ctx, span, log), RequestLogAvailability.COMPLETE);
+
+        try (SpanInScope ignored = tracer.withSpanInScope(span)) {
+            return delegate().execute(ctx, req);
         }
+    }
 
-        if (spanId == null) {
-            headers.add(SAMPLED, "0");
-        } else {
-            headers.add(SAMPLED, "1");
-            headers.add(TRACE_ID, IdConversion.convertToString(spanId.traceId));
-            headers.add(SPAN_ID, IdConversion.convertToString(spanId.spanId));
-            if (!spanId.root()) {
-                headers.add(PARENT_SPAN_ID, IdConversion.convertToString(spanId.parentId));
-            }
+    private void finishSpan(ClientRequestContext ctx, Span span, RequestLog log) {
+        final Throwable cause = log.responseCause();
+        final String clientResultText = cause == null ? "success" : "failure";
+        span.tag("client.uri", log.scheme().uriText() + "://" + log.host() + ctx.path() + '#' + log.method())
+            .tag("client.result", clientResultText);
```

Comment:
- clientResultText could be inlined.
- could append `'?' + ctx.query()` if `ctx.query()` is non-null
- we could use RPC method name instead of `log.method()` if RPC method name is available.